src/App/Fossa/Config/Project/Edit.hs

```diff
@@ -0,0 +1,135 @@
+    <*> optional (strOption (applyFossaStyle <> long "team" <> short 'T' <> stringToHelpDoc "The team to associate with the project."))
+    <*> parsePolicyOptions
+    <*> many (strOption (applyFossaStyle <> long "project-label" <> stringToHelpDoc "Assign up to 5 labels to the project."))
+
+mergeOpts ::
+  ( Has Diagnostics sig m
+  ) =>
+  Maybe ConfigFile ->
+  EnvVars ->
+  EditOpts ->
+  m EditConfig
+mergeOpts maybeConfig envVars cliOpts@EditOpts{..} = do
+  let maybeProjectLocator = maybe (projectLocatorOpts) (mergeProjectLocator projectLocatorOpts) maybeConfig
+  apiOpts <- collectApiOpts maybeConfig envVars cliOpts
+  projectMetadataRevision <- maybe (pure projectMetadataRevisionOpts) (mergeProjectMetadata projectMetadataRevisionOpts) maybeConfig
+  case maybeProjectLocator of
+    Just projectLocator' | not (Data.Text.null projectLocator') -> pure $ EditConfig apiOpts projectLocator' projectMetadataRevision
+    _ -> fatalText "A project locator needs to be specified to edit a FOSSA project"
+
+collectApiOpts :: (Has Diagnostics sig m) => Maybe ConfigFile -> EnvVars -> EditOpts -> m ApiOpts
+collectApiOpts maybeConfig envVars EditOpts{..} = do
+  apikey <- validateApiKeyGeneric maybeConfig (envApiKey envVars) apiKeyOpts
+  let configUri = maybeConfig >>= configServer >>= mkURI
+      baseuri = baseUrlOpts <|> configUri
+  pure $ ApiOpts baseuri apikey defaultApiPollDelay
+
+mergeProjectLocator :: Maybe Text -> ConfigFile -> Maybe Text
+mergeProjectLocator maybeProjectLocator config = maybeProjectLocator <> (configProject config >>= configProjLocator)
+
+mergeProjectMetadata :: Has Diagnostics sig m => ProjectMetadataRevision -> ConfigFile -> m ProjectMetadataRevision
+mergeProjectMetadata ProjectMetadataRevision{..} config =
+  case (projectPolicyRevision, cfgPolicy) of
+    (Just (PolicyId _), Just (PolicyName _)) -> err
+    (Just (PolicyName _), Just (PolicyId _)) -> err
+    _ -> pure . mkMeta $ projectPolicyRevision <|> cfgPolicy
+  where
+    err = fatalText "Only one of policy or policyId can be set. Check your cli options and .fossa.yml to ensure you aren't specifying both."
+    cfgPolicy = configProject config >>= configPolicy
+
+    mkMeta :: Maybe Policy -> ProjectMetadataRevision
+    mkMeta policy =
+      ProjectMetadataRevision
+        { projectTitleRevision = projectTitleRevision <|> (configProject config >>= configName)
+        , projectUrlRevision = projectUrlRevision <|> (configProject config >>= configUrl)
+        , projectJiraKeyRevision = projectJiraKeyRevision <|> (configProject config >>= configJiraKey)
+        , projectLinkRevision = projectLinkRevision <|> (configProject config >>= configLink)
+        , projectTeamRevision = projectTeamRevision <|> (configProject config >>= configTeam)
+        , projectPolicyRevision = policy
+        , projectLabelRevision = projectLabelRevision <|> (maybe [] configLabel (configProject config))
+        }
```

Comment:
[Optional] It's a bit more readable if you make a closure like:
```haskell
configVal f = configProject config >>= f
```
Then a snippet like:
```haskell
      { projectTitleRevision = projectTitleRevision <|> (configProject config >>= configName)
        , projectUrlRevision = projectUrlRevision <|> (configProject config >>= configUrl)
        , projectJiraKeyRevision = projectJiraKeyRevision <|> (configProject config >>= 
```
is:
```haskell
      { projectTitleRevision = projectTitleRevision <|> configVal configName
        , projectUrlRevision = projectUrlRevision <|> configVal configUrl
        , projectJiraKeyRevision = projectJiraKeyRevision <|> configVal configJiraKey
        ...
```